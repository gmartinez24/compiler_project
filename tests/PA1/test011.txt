//Test to see if nested paren work, as well as nested functions
main

int output;


function subtractOne (int x) : int
{
    int i;
    i = x - 1;
    return i;
};

{
    int input = 1;
    /*

    /**/ //Testing semi-nested block comment
    call printLn(subtractOne(input));
}.