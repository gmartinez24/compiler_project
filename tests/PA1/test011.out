MAIN
INT
IDENT	output
SEMICOLON
FUNC
IDENT	subtractOne
OPEN_PAREN
INT
IDENT	x
CLOSE_PAREN
COLON
INT
OPEN_BRACE
INT
IDENT	i
SEMICOLON
IDENT	i
ASSIGN
IDENT	x
SUB
INT_VAL	1
SEMICOLON
RETURN
IDENT	i
SEMICOLON
CLOSE_BRACE
SEMICOLON
OPEN_BRACE
INT
IDENT	input
ASSIGN
INT_VAL	1
SEMICOLON
CALL
IDENT	printLn
OPEN_PAREN
IDENT	subtractOne
OPEN_PAREN
IDENT	input
CLOSE_PAREN
CLOSE_PAREN
SEMICOLON
CLOSE_BRACE
PERIOD
EOF
